lib/oli_web/live/collaboration_live/collab_space_view.ex

```diff
@@ -266,18 +249,16 @@ defmodule OliWeb.CollaborationLive.CollabSpaceView do
   end
 
   # -----------Edit Post-----------#
+  def handle_event("set_editing_post", %{"post_id" => post_id}, socket) do
+    editing_post = socket.assigns.editing_post
 
-  def handle_event("display_edit_modal", %{"id" => id}, socket) do
-    post = Collaboration.get_post_by(%{id: id})
-    changeset = Collaboration.change_post(post)
-
-    modal_assigns = %{
-      id: "edit_post_modal",
-      on_submit: "edit_post",
-      changeset: changeset
-    }
+    {is_edition_mode, post} =
+      case editing_post && Integer.to_string(editing_post.id) == post_id do
+        true -> {false, nil}
+        _ -> {true, Collaboration.get_post_by(%{id: post_id})}
+      end
 
-    display_post_modal(modal_assigns, assign(socket, editing_post: post))
+    {:noreply, assign(socket, editing_post: post, is_edition_mode: is_edition_mode)}
```

Comment:
I noticed than when we enter the "edit" mode, we do not have focus set to the corresponding textArea. To do so, we can push an event:
```suggestion
    socket = assign(socket, editing_post: post, is_edition_mode: is_edition_mode)

    {:noreply, push_event(socket, "set_focus", %{id: "post_text_area_#{post_id}"})}
```

```
// textarea_listener.ts

export const TextareaListener = {
  mounted(): void {
   ...
   window.addEventListener(`phx:set_focus`, (e: any) => {
      const el = document.getElementById(e.detail.id);
      if (el) {
        el.focus();
      }
    });
  },
...
};
```

```
// show.ex

<TextArea id={"post_text_area_#{@post.id}"}
              opts={
                placeholder: "Reply",
                "data-grow": "true",
                "data-initial-height": 44,
                onkeyup: "resizeTextArea(this)"
              }
              class="torus-input border-r-0 collab-space__reply"
            />

```

WDYT?

Reply:
Thanks for this ❤️